activerecord/lib/active_record/relation.rb

```diff
@@ -22,10 +22,11 @@ class Relation
     alias :loaded? :loaded
     alias :locked? :lock_value
 
-    def initialize(klass, table, predicate_builder, values = {})
+    def initialize(klass, table, predicate_builder, values = {}, connection_name = nil)
```

Comment:
I had that originally - see the first commit.  Lots of test failures relating to Marshal.dump.